src/_pytest/mark/structures.py

```diff
@@ -282,12 +283,17 @@ class MarkInfo(object):
     """ Marking object created by :class:`MarkDecorator` instances. """
 
     _marks = attr.ib()
-    combined = attr.ib(
-        repr=False,
-        default=attr.Factory(
-            lambda self: reduce(Mark.combined_with, self._marks), takes_self=True
-        ),
-    )
+    combined = attr.ib(repr=False)
+
+    @combined.default
+    def _combined_default(self):
```

Comment:
this is ABSOLUTELY not the place not fix - in addition the python version check makes absolutely no sense

`_marks` should get a validator that requires its a list of mark objects

Reply:
No need to shout. ;)